gen/bind_generator.jl

```diff
@@ -4,29 +4,26 @@ using Base.Meta: isexpr, quot
 # of the translations explicitly.
 const bind_exceptions = Dict{Symbol,Symbol}()
 
-# Version-numbered identifiers
-push!(bind_exceptions, :h5a_create => :H5Acreate2)
-push!(bind_exceptions, :h5d_create => :H5Dcreate2)
-push!(bind_exceptions, :h5d_open => :H5Dopen2)
-push!(bind_exceptions, :h5e_get_auto => :H5Eget_auto2)
-push!(bind_exceptions, :h5e_set_auto => :H5Eset_auto2)
-push!(bind_exceptions, :h5g_create => :H5Gcreate2)
-push!(bind_exceptions, :h5g_open => :H5Gopen2)
-push!(bind_exceptions, :h5o_get_info => :H5Oget_info1)
-push!(bind_exceptions, :h5p_get_filter_by_id => :H5Pget_filter_by_id2)
-push!(bind_exceptions, :h5r_dereference => :H5Rdereference2)
-push!(bind_exceptions, :h5r_get_obj_type => :H5Rget_obj_type2)
-push!(bind_exceptions, :h5t_array_create => :H5Tarray_create2)
-push!(bind_exceptions, :h5t_commit => :H5Tcommit2)
-push!(bind_exceptions, :h5t_get_array_dims => :H5Tget_array_dims2)
-push!(bind_exceptions, :h5t_open => :H5Topen2)
-
 # Distinguishes 32-bit vs 64-bit handle arguments
 push!(bind_exceptions, :h5p_get_fapl_mpio32 => :H5Pget_fapl_mpio)
 push!(bind_exceptions, :h5p_get_fapl_mpio64 => :H5Pget_fapl_mpio)
 push!(bind_exceptions, :h5p_set_fapl_mpio32 => :H5Pset_fapl_mpio)
 push!(bind_exceptions, :h5p_set_fapl_mpio64 => :H5Pset_fapl_mpio)
 
+# An expression which is injected at the beginning of the API defitions to aid in doing
+# (pre)compile-time conditional compilation based on the libhdf5 version.
+_libversion_build_expr = quote
+    _libversion_build = let
```

Comment:
Maybe rename this to something that denotes it refers to the libhdf5 library, e.g.
```
_libhdf5_build_ver
```